src/snowflake/cli/api/project/schemas/entities/snowpark_entity.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Dict, List, Literal, Optional, Union
+
+from pydantic import Field, field_validator
+from snowflake.cli.api.project.schemas.entities.common import EntityModelBase
+from snowflake.cli.api.project.schemas.identifier_model import ObjectIdentifierModel
+from snowflake.cli.api.project.schemas.snowpark.argument import Argument
+from snowflake.cli.api.project.schemas.updatable_model import DiscriminatorField
+
+
+class SnowparkEntityModel(EntityModelBase):
```

Comment:
Aren't we missing an entity type here?
```suggestion
class SnowparkEntityModel(EntityModelBase):
    type: Literal["snowpark"] = DiscriminatorField()  # noqa: A003
```

Reply:
In what context `snowpark` would be used? Currently it's a base class for common elements between entities.

Reply:
Got it. For clarity, maybe we should have `SnowparkEntityModel` to also inherit from `ABC` and add "Base" to its name?
`SnowparkFunctionEntity` might also be more accurate.